test/ngAnimate/animateRunnerSpec.js

```diff
@@ -0,0 +1,251 @@
+      expect(status2).toBeFalsy();
+
+      $rootScope.$digest();
+      expect(status1).toBe(expectedStatus);
+      expect(status2).toBe(expectedStatus);
+    });
+  });
+
+  it("should expose/create the contained promise when getPromise() is called",
+    inject(function($animateRunner, $rootScope) {
+
+    var runner = new $animateRunner();
+    expect(isPromiseLike(runner.getPromise()));
+  }));
+
+  describe(".all()", function() {
+    it("should resolve when all runners have naturally resolved",
+      inject(function($$rAF, $animateRunner) {
+
+      var runner1 = new $animateRunner();
+      var runner2 = new $animateRunner();
+      var runner3 = new $animateRunner();
+
+      var status;
+      $animateRunner.all([runner1, runner2, runner3], function(response) {
+        status = response;
+      });
+
+      runner1.complete(true);
+      runner2.complete(true);
+      runner3.complete(true);
+
+      expect(status).toBeUndefined();
+
+      $$rAF.flush();
+
+      expect(status).toBe(true);
+    }));
+
+    they("should immediately resolve if and when all runners have resolved been $prop",
+      { ended: 'end', cancelled: 'cancel' }, function(method) {
+
+      inject(function($$rAF, $animateRunner) {
+        var runner1 = new $animateRunner();
+        var runner2 = new $animateRunner();
+        var runner3 = new $animateRunner();
+
+        var expectedStatus = method === 'end' ? true : false;
+
+        var status;
+        $animateRunner.all([runner1, runner2, runner3], function(response) {
+          status = response;
+        });
+
+        runner1[method]();
+        runner2[method]();
+        runner3[method]();
+
+        expect(status).toBe(expectedStatus);
+      });
+    });
+
+    it("should return a status of `false` if more than one runner was cancelled",
+      inject(function($$rAF, $animateRunner) {
+
+      var runner1 = new $animateRunner();
+      var runner2 = new $animateRunner();
+      var runner3 = new $animateRunner();
+
+      var status;
+      $animateRunner.all([runner1, runner2, runner3], function(response) {
+        status = response;
+      });
+
+      runner1.end();
+      runner2.end();
+      runner3.cancel();
+
+      expect(status).toBe(false);
+    }));
+  });
+
+  describe(".chain()", function() {
+    it("should evaluate an array of functions in a chain",
+      inject(function($$rAF, $animateRunner) {
+
+      var runner1 = new $animateRunner();
+      var runner2 = new $animateRunner();
+      var runner3 = new $animateRunner();
+
+      var log = [];
+
+      var items = [];
+      items.push(function(fn) { log.push(1); runner1.done(fn); });
+      items.push(function(fn) { log.push(2); runner2.done(fn); });
+      items.push(function(fn) { log.push(3); runner3.done(fn); });
+
+      var status;
+      $animateRunner.chain(items, function(response) {
+        status = response;
+      });
+
+      runner1.complete(true);
+      runner2.complete(true);
+      runner3.complete(true);
```

Comment:
This does not make it clear (nor tests) what happens if the runners are complete in a different order than the order they appear in `items`.

Would it be better/more explicit to `complete()` them in a different order and verify that the log order still matches their order in `items` ?